### ISSUE TITLE
Undefined indexes with login backwards compat during purchase

### ISSUE THREAD

**Issue by thomasgriffin** [easydigitaldownloads/easy-digital-downloads#5281]
We've recently implemented Sentry for bug tracking in our applications, and we've come up with one over and over again in EDD. Here are the relevant details from Sentry:

```
ErrorException: Undefined index: user_pass
  File "/dom9512/wp-content/plugins/easy-digital-downloads/includes/process-purchase.php", line 754, in handleError
    edd_log_user_in( $user['user_id'], $user['user_login'], $user['user_pass'] );
  File "/dom9512/wp-content/plugins/easy-digital-downloads/includes/process-purchase.php", line 754, in edd_get_purchase_form_user
    edd_log_user_in( $user['user_id'], $user['user_login'], $user['user_pass'] );
  File "/dom9512/wp-content/plugins/easy-digital-downloads/includes/process-purchase.php", line 48, in edd_process_purchase_form
    $user = edd_get_purchase_form_user( $valid_data );
  File "/wordpress-4.6.1/wp-includes/plugin.php", line 524, in do_action
    call_user_func_array($the_['function'], array_slice($args, 0, (int) $the_['accepted_args']));
  File "/dom9512/wp-content/plugins/easy-digital-downloads/includes/actions.php", line 40, in edd_post_actions
    do_action( 'edd_' . $_POST['edd_action'], $_POST );
...
(6 additional frame(s) were not displayed)

Undefined index: user_pass
```

I've traced it down to this occurring when a logged-in user adds something to the cart and attempts to purchase, but then for whatever reason leaves the page. They come back to the page at some point later in time and attempt to complete the purchase. However, at this point, the session has been cleared out from the cart and EDD thinks they are a guest user. Here is the relevant `$valid_data` variable output given by Sentry (it is consistent across all products):

```
{
	cc_info: {
		card_address: , 
		card_address_2: , 
		card_city: , 
		card_country: , 
		card_cvc: , 
		card_exp_month: , 
		card_exp_year: , 
		card_name: , 
		card_number: , 
		card_state: , 
		card_zip: 
	}, 
	discount: , 
	gateway: stripe, 
	guest_user_data: [], 
	logged_user_data: [], 
	login_user_data: {
		user_id: -1
	}, 
	need_new_user: false, 
	need_user_login: true, 
	new_user_data: []
}
```

Basically, at that line (around 760-ish or `process-purchase.php`) there needs to be a check to ensure that login data actually exists before attempting to login. If it doesn't exist, I think the likely course of action would be to redirect back to the checkout with an error.

**Comment by pippinsplugins** [comment]
Thanks for the report! Should be pretty simple to resolve.

**Comment by pippinsplugins** [comment]
Thanks @benjaminprojas, we'll get that tested as soon as we can.

**Comment by sunnyratilal** [comment]
Works fine for me.

**Comment by sunnyratilal** [comment]
PR at #5303 for ref

**Comment by sunnyratilal** [comment]
New PR is at #5355 as we needed to sync the PR with `release/2.7` in order for unit tests to run.

**Comment by sunnyratilal** [comment]
Merged into `release/2.7`
